src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -35,51 +35,85 @@
 				"%itemtype/inventorytype% (named|with name[s]) %string%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<String> name;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[0]);
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't create a '" + literal.toString() + "' inventory. It's not creatable!");
+				return false;
+			}
+		}
 		return true;
 	}
-	
+
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		String name = this.name.getSingle(e);
+	protected Object[] get(Event event, Object[] source) {
+		String name = this.name.getSingle(event);
 		if (name == null)
-			return get(source, obj -> obj); // No name provided, do nothing
+			return get(source, object -> {
+				if (object instanceof InventoryType) {
+					try {
+						return Bukkit.createInventory(null, (InventoryType) object);
+					} catch (NullPointerException | IllegalArgumentException e) {
+						// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+						// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+						// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+						Skript.error("You can't create a '" + Classes.toString((InventoryType) object) + "' inventory. It's not creatable!");
+						return null;
+					}
+				}
+				return object; // Return the same ItemType they passed without applying a name.
+			});
 		return get(source, new Getter<Object, Object>() {
 			@Override
 			@Nullable
-			public Object get(Object obj) {
-				if (obj instanceof InventoryType)
-					return Bukkit.createInventory(null, (InventoryType) obj, name);
-				if (obj instanceof ItemStack) {
-					ItemStack stack = (ItemStack) obj;
+			public Object get(Object object) {
+				if (object instanceof InventoryType) {
+					try {
+						return Bukkit.createInventory(null, (InventoryType) object, name);
+					} catch (NullPointerException | IllegalArgumentException e) {
+						// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+						// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+						// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+						Skript.error("You can't create a '" + Classes.toString((InventoryType) object) + "' inventory. It's not creatable!");
+						return null;
+					}
+				}
+				if (object instanceof ItemStack) {
```

Comment:
I don't know why this is being checked when itemstacks are not being admitted by the expression, I believe that in order to avoid itemstacks somehow being passed, the norm is to do something like this on init:
```
public boolean init(...) {
   setExpr(exprs[0].getConvertedExpression(InventoryType.class));

   if (getExpr() == null)
      setExpr(exprs[0].getConvertedExpression(ItemType.class));
}
```

That said, I still find weird that such checks are necessary, is this a side-effect of expressions that take more than one type? Does this also happen to pass ItemStacks if you accept ItemType alone?

Reply:
See https://github.com/SkriptLang/Skript/issues/1822 for all questions. It's existing implementation.